recipe_scrapers/monsieurcuisine.py

```diff
@@ -1,72 +1,122 @@
-        if container and container.has_attr("src"):
-            container = container["src"]
-
-        if not container:
-            return None
-
-        return "https://www." + self.host() + container
+        return self.data.get("data").get("recipe").get("thumbnail").get("landscape")
 
     def ingredients(self):
-        container = self.soup.find(
-            "div", {"class": "recipe--ingredients-html-item col-md-8"}
+        ingredients = []
+        raw_ingridients = (
+            self.data.get("data")
+            .get("recipe")
+            .get("servingSizes")[0]
+            .get("ingredients")
         )
+        for raw_ingredient in raw_ingridients:
+            ingredients.append(
+                f"{raw_ingredient.get('amount')} {raw_ingredient.get('unit')} {raw_ingredient.get('name')}"
+            )
 
-        ingredients = container.findAll("li")
-
-        return [normalize_string(ingredient.get_text()) for ingredient in ingredients]
+        return ingredients
 
     def instructions(self):
-        container = self.soup.find("div", {"class": "recipe--instructions"})
+        instruction = (
+            self.data.get("data")
+            .get("recipe")
+            .get("servingSizes")[0]
+            .get("instruction")
+        )
 
-        instructions = container.find_all("li")
+        instruction = re.sub(r"(\r\n\r\n)|(\n\n)", "__DOUBLE_NEWLINE__", instruction)
 
-        return "\n".join(
-            [normalize_string(instruction.get_text()) for instruction in instructions]
-        )
+        instruction = re.sub(r"(\r\n)|(\n)", " ", instruction)
 
-    def ratings(self):
-        container = self.soup.find("div", {"class": "recipe--rating"})
+        instruction = instruction.replace("__DOUBLE_NEWLINE__", "\n")
 
-        full_stars = container.find_all("span", {"class": "mc-star_filled"})
-        half_stars = container.find_all("span", {"class": "mc-half_star_thick"})
-        return len(full_stars) + 0.5 * len(half_stars)
+        return instruction
+
+    def ratings(self):
+        return self.data.get("data").get("recipe").get("rating")
+
+    def nutrients(self):
+        nutrients = self.data.get("data").get("recipe").get("nutrients")
+
+        for nutrient in nutrients:
+            if nutrient.get("name") == "Calories":
+                calories = nutrient
+            if nutrient.get("name") == "Carbohydrate":
+                carbohydrates = nutrient
+            if nutrient.get("name") == "Fat":
+                fat = nutrient
+            if nutrient.get("name") == "Protein":
+                protein = nutrient
+
+        return {
+            "calories": f'{calories.get("amount")} {calories.get("unit")}',
+            "fatContent": f'{fat.get("amount")} {fat.get("unit")}',
+            "carbohydrateContent": f'{carbohydrates.get("amount")} {carbohydrates.get("unit")}',
+            "proteinContent": f'{protein.get("amount")} {protein.get("unit")}',
+        }
```

Comment:
Reduce the chance for copy/paste typos by refactoring this code:

```suggestion
    def nutrients(self):
        schema_mapping = {
            "Calories": "calories",
            "Carbohydrate": "carbohydrateContent",
            "Fat": "fatContent",
            "Protein": "proteinContent",
        }

        results = {}
        for nutrient in self.data.get("data").get("recipe").get("nutrients"):
            output_field = schema_mapping.get(nutrient.get("name"))
            if output_field is not None:
                results[output_field] = f'{nutrient.get("amount")} {nutrient.get("unit")}'
        return results
```